### ISSUE TITLE
Fused MoE RoutingMethodType has no sqrtsoftplus scoring — DeepSeek-V4.1 cannot use the internal-routing trtllm path

### ISSUE THREAD

**Issue by Tushar-ml** [flashinfer-ai/flashinfer#5190]
## Summary

`trtllm_fp4_block_scale_moe` computes routing inside the kernel, selected from a fixed `RoutingMethodType` enum. All ten current methods score with **softmax**, **sigmoid**, or plain **top-k**:

| # | method | scoring |
|---|---|---|
| 0 | Default | Softmax -> TopK |
| 1 | Renormalize | TopK -> Softmax |
| 2 | DeepSeekV3 | Sigmoid -> BiasAdd -> Top2-in-group -> Top4 groups -> TopK |
| 3 | Llama4 | Top1 -> Sigmoid |
| 4 | RenormalizeNaive | Softmax -> TopK -> Renormalize |
| 5 | TopK | TopK only |
| 6 | SigmoidRenorm | Sigmoid -> TopK -> Renormalize |
| 7 | MiniMax2 | Sigmoid+Bias -> TopK -> ScaledSumNormalize |
| 8 | Sigmoid | Sigmoid -> TopK |
| 9 | TopKSigmoid | TopK -> Sigmoid |

**DeepSeek-V4.1 scores with `sqrtsoftplus`**, which is none of these.

## Why this matters

`deepseek-ai/DeepSeek-V4.1-Flash` (`config.json`, `text_config`):

```
scoring_func          = sqrtsoftplus
topk_method           = noaux_tc
n_routed_experts      = 384
num_experts_per_tok   = 6
n_group               = None
topk_group            = None
routed_scaling_factor = 1.5
norm_topk_prob        = True
```

Because no enum value reproduces this, the model can only use `trtllm_fp4_block_scale_routed_moe`, supplying pre-computed expert assignments. Selecting a different `RoutingMethodType` anyway would not be a workaround — it would silently select **different experts**, i.e. a correctness bug rather than a config mismatch.

In practice the first symptom is misleading. SGLang passes `RoutingMethodType.DeepSeekV3` for this family, so users see:

```
trtllm_fused_moe_runner.cu:155
FLASHINFER_CHECK(topkGroup <= 4, "For DeepSeek routing method, must have topkGroup <= 4")
```

which reads like a grouping/config problem, when the actual blocker is the unsupported scoring function.

## Request

Either:

1. add a `sqrtsoftplus` routing method (softplus-then-sqrt scoring, ungrouped top-k with correction bias and renormalization), or
2. document that models whose scoring is outside the enum must use the routed variant — and note the consequence, that they lose deferred finalize (see the companion issue filed against SGLang).

Even just (2) would save people the misdiagnosis: the `topkGroup` error points at the wrong thing.

## Environment

- FlashInfer 0.6.18, SGLang `feat/decode-first-on-dsv41` @ a70d0d46
- 4x NVIDIA B200 (SM100), driver 580.173.02, CUDA 13.0
- `AtomicChat/DeepSeek-V4.1-Flash-NVFP4-nvidia`, TP4

**Comment by 200lz** [comment]
I’d like to take the FlashInfer core routing slice for this.

My first step would be to pin the DSV4.1 routing contract against the current SGLang implementation with a small PyTorch reference: `sqrt(softplus(logits))`, correction-bias expert selection, ungrouped top-k, unbiased routed weights, renormalization, and routed scaling.

Before adding CUDA code, I’ll use close-score adversarial vectors where sigmoid / softmax / sqrtsoftplus produce different expert rankings, so the tests catch semantic misrouting rather than only numerical drift.

Then I’ll check whether the existing `routingCustom` preprocess / selection / postprocess machinery can express this cleanly. If it can, I’d prefer the smallest semantic primitive (e.g. `SqrtSoftplus`) rather than unnecessarily coupling the enum to a model name, and append the enum value without renumbering existing methods. If biased selection with unbiased final weights needs a separate score contract, I’ll keep that handling scoped rather than changing DeepSeekV3 behavior.

Happy to align on whether maintainers prefer an algorithmic `SqrtSoftplus` method or a `DeepSeekV4`-named routing method before wiring the public enum.
